src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -3248,6 +3249,16 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
 
             break;
         }
+        case SMART_ACTION_SET_WORLD_STATE:
+        {
+            sWorld->setWorldState(e.action.worldState.index, e.action.worldState.value);
```

Comment:
Maybe it's idea for follow up PR but it looks to me like it would be nice to refactor sWorld and move world state related logic to the new class.

Reply:
Yes, it should be more centralized. The functions from the World instance  should be moved elsewhere

these 2 from `World.h`:
```
virtual void setWorldState(uint32 index, uint64 value) = 0;
[[nodiscard]] virtual uint64 getWorldState(uint32 index) const = 0;
```